tethys_cli/db_commands.py

```diff
@@ -0,0 +1,143 @@
+                           superuser_name='tethys_super', superuser_password='pass',
+                           portal_superuser_name='admin', portal_superuser_email='', portal_superuser_password='pass')
+
+
+def init_db_server(db_dir=None, **kwargs):
+    args = ['initdb', '-U', 'postgres', '-D', f'{db_dir}/data']
+    run_process(args)
+
+
+def start_db_server(db_dir=None, port=None, **kwargs):
+    args = ['pg_ctl', '-U', 'postgres', '-D', f'{db_dir}/data', '-l', f'{db_dir}/logfile', 'start', '-o', f'-p {port}']
+    run_process(args)
+
+
+def stop_db_server(db_dir=None, **kwargs):
+    args = ['pg_ctl', '-U', 'postgres', '-D', f'{db_dir}/data', 'stop']
+    run_process(args)
+
+
+def create_db_user(port=None, username=None, password=None, db_name=None, is_superuser=False, **kwargs):
+    db_name = db_name or username
+
+    if is_superuser:
+        command = f"CREATE USER {username} WITH CREATEDB NOCREATEROLE SUPERUSER PASSWORD '{password}';"
+    else:
+        command = f"CREATE USER {username} WITH NOCREATEDB NOCREATEROLE NOSUPERUSER PASSWORD '{password}';"
+
+    args = ['psql', '-U', 'postgres', '-p', f'{port}', '--command', command]
+    run_process(args)
+    if not is_superuser:
+        args = ['createdb', '-U', 'postgres', '-p', f'{port}', '-O', username, db_name, '-E', 'utf-8', '-T',
+                'template0']
+        run_process(args)
+
+
+def create_tethys_db(port=None, db_name=None, username=None, password=None,
+                     superuser_name=None, superuser_password=None, **kwargs):
+    create_db_user(port=port, username=username, password=password, db_name=db_name)
+    if superuser_name is not None and superuser_password is not None:
+        create_db_user(port=port, username=superuser_name, password=superuser_password, is_superuser=True)
+
+
+def sync_tethys_db(db_alias=None, **kwargs):
+    manage_path = get_manage_path(None)
+    db_alias = db_alias or 'default'
+    args = ['python', manage_path, 'migrate', '--database', db_alias]
+    run_process(args)
+
+
+def create_portal_superuser(portal_superuser_name='admin', portal_superuser_email='', portal_superuser_password='pass',
+                            **kwargs):
+    load_apps()
+    from django.contrib.auth.models import User  # noqa: E402
+    User.objects.create_superuser(portal_superuser_name, portal_superuser_email, portal_superuser_password)
+
+
+def configure_tethys_db(**kwargs):
+    init_db_server(**kwargs)
+    start_db_server(**kwargs)
+    create_tethys_db(**kwargs)
+    sync_tethys_db(**kwargs)
+    create_portal_superuser(**kwargs)
+
+
+def process_args(args):
+    db_settings = settings.DATABASES[args.db_alias]
+    db_dir = None
+    if args.command in ['init', 'start', 'stop', 'configure']:
+        try:
+            db_dir = db_settings['DIR']
+        except KeyError:
+            raise RuntimeError(f'The tethys db {args.command} command can only be used with local databases.')
```

Comment:
What would it take to make this work on remote postgres databases?

Reply:
The `init`, `start`, and `stop` commands don't make sense for a remote database. Those commands are for setting up and controlling a database server. If you already have a remote database server running then you don't need to run these commands.

The `configure` command is a shortcut for running `init`, `start`, `create`, `migrate`, and `createsuperuser`. I could add an option to have the shortcut without the first two commands for a remote database. Do you think that would be useful?

Reply:
I like the idea of having an option that would skip the first three commands. `tethys db configure --remote`? Or perhaps we just have it do that automatically if the database host is not localhost or 127.0.0.1

Reply:
I've made it so that `configure` will only run `init` and `start` if `db_dir` is not `None`.  That will allow `configure` to work by default wether or not it's remote.

Reply:
@sdc50 won’t this still throw an exception if DIR isn’t in the settings.py DATABASES setting?

Reply:
Ah, yes. I need to remove 'configure' from the list on line 110.